common/serial_reader.h

```diff
@@ -0,0 +1,104 @@
+// USB-reader
+#pragma once
+
+#include <string>
+#include "termios.h"
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <fstream>
+#include <unistd.h>
+#include <stdio.h>
+#include <sys/ioctl.h>
+#include <chrono>
+#include <thread>
+#include <errno.h>
+
+#include <libserialport.h> // cross platform serial port lib
+
+
+#define BAUDRATE B9600
+ 
+
+class SerialReader {
+    private:
+    int fileDescriptor;
+    bool connected = false;
+   
+    struct termios settings;
+
+
+    public:
+    SerialReader() {}
```

Comment:
You should initialise ``fileDescriptor`` to ``-1``, because otherwise the user could call ``readData()`` before it's set